### DISCUSSION TITLE
Keycloak 21.1.1 with Java Keycloak Admin Client always returns HTTP 404

### DISCUSSION THREAD

**Discussion by dmestudent** [keycloak/keycloak#22078] [Q&A]
Hello,

I never had problems with using the Keycloak Admin Client library, but now after upgrading Keycloak to 21.1.1 I can not get it to work anymore. 

I create my bean like this:

```
    public Keycloak keycloak() {
        Keycloak keycloak = KeycloakBuilder.builder()
                .grantType(CLIENT_CREDENTIALS)
                .serverUrl("http://docker-dev.global:10500")
                .realm("testrealm")
                .clientId("test-service")
                .clientSecret("redacted")
                .build();

        return keycloak;
    }
```

I created a realm called "testrealm" and a client called "test-service". The client is configured with "Client authentication" turned on and has a "service account roles" turned on.

I granted the service account ALL realm-management roles + the default roles of the client. 

I tested retrieving the access_token using the clientId + secret using Postman without any problem. I also checked that my java app receives the access_token properly. 

```
        // This one returns the proper access_token, checking the access token using jwt.io reveals that all roles are properly present in the token
        AccessTokenResponse at = keycloak.tokenManager().getAccessToken();

        // Calling this will result the the following error 
        List<UserRepresentation> users = keycloak.realm("testrealm").users().list();
```

Error: javax.ws.rs.NotFoundException: HTTP 404 Not Found

I don`t know how I should proceed from here. The error message does not give any information whats going wrong. There is no information in the Keycloak log either. I have been trying various things for hours now. Any help is highly appreciated.

Thanks in advance.

EDIT: By switching to username and password authentication, creating a user in the realm which also got granted all "realm-management" roles, setting "admin-cli" as client and calling the functions above everything works as expected. 

Could it be that there is a bug in the Keycloak Admin Client library 21.1.1 regarding using confidential clientId + secret authentictation?

**Comment by vmuzikar** [comment]
@dmestudent Please create an Issue with as specific steps to reproduce as possible. Thanks.
